R/log_shiny_input_changes.R

```diff
@@ -37,35 +37,36 @@ log_shiny_input_changes <- function(
     input,
     namespace = NA_character_,
     excluded_inputs = character(),
-    excluded_pattern = "_width$") {
-  session <- shiny::getDefaultReactiveDomain()
+    excluded_pattern = "_width$",
+    session = shiny::getDefaultReactiveDomain()) {
   if (!(shiny::isRunning() || inherits(session, "MockShinySession") || inherits(session, "session_proxy"))) {
     stop("No Shiny app running, it makes no sense to call this function outside of a Shiny app")
   }
-  ns <- ifelse(!is.null(session), session$ns(character(0)), "")
 
-  # utils::assignInMyNamespace and utils::assignInNamespace are needed
-  # so that observer is executed only once, not twice.
-  input_values <- shiny::isolate(shiny::reactiveValuesToList(input))
-  utils::assignInMyNamespace("shiny_input_values", input_values)
+  if (logger::TRACE > logger::as.loglevel(get_val("TEAL.LOG_LEVEL", "teal.log_level", "INFO"))) {
```

Comment:
**Very important condition** to avoid setting `observers` if `log level < TRACE`. `input` object have so many bindings that observing `reactiveValuesToList(input)` is ridiculous. This applies the most to the DT objects which have a huuuuge-amount of shiny bindings set and they might change every tab switch (some are returning datetime value).

Reply:
Duly noted!